docs/src/main/sphinx/functions/json.rst

```diff
@@ -1163,6 +1166,269 @@ id         child
 103        'missing'
 ========== ================
 
+.. _json_array:
+
+json_array
+----------
+
+The ``json_array`` function creates a JSON array containing given elements.
+
+.. code-block:: text
+
+    JSON_ARRAY(
+        [ array_element [, ...]
+          [ { NULL ON NULL | ABSENT ON NULL } ] ],
+        [ RETURNING type [ FORMAT JSON [ ENCODING { UTF8 | UTF16 | UTF32 } ] ] ]
+        )
+
+Argument types
+^^^^^^^^^^^^^^
+
+The array elements can be arbitrary expressions. Each passed value is converted
+into a JSON item according to its type and optional ``FORMAT`` and ``ENCODING``
+specification.
+
+You can pass SQL values of types boolean, numeric, and character string. They
+are converted to corresponding JSON literals:
+
+.. code-block:: text
+
+    SELECT json_array(true, 12e-1, 'text') --> '[true,1.2,"text"]'
+
+Additionally to SQL values, you can pass JSON values. They are character or
+binary strings with specified format and optional encoding:
+
+.. code-block:: text
+
+    SELECT json_array(
+                      '[  "text"  ] ' FORMAT JSON,
+                      X'5B0035005D00' FORMAT JSON ENCODING UTF16
+                     )
+    --> '[["text"],[5]]'
+
+You can also nest other json-returning functions. In that case, the ``FORMAT``
+option is implicit:
+
+.. code-block:: text
+
+    SELECT json_array(json_query('{"key" : [  "value"  ]}', 'lax $.key'))
+    --> '[["value"]]'
+
+Other passed values are cast to varchar, and they become JSON text literals:
+
+.. code-block:: text
+
+    SELECT json_array(
+                      DATE '2001-01-31',
+                      UUID '12151fd2-7586-11e9-8f9e-2a86e4085a59'
+                     )
+    --> '["2001-01-31","12151fd2-7586-11e9-8f9e-2a86e4085a59"]'
+
+You can omit the arguments altogether to get an empty array:
+
+.. code-block:: text
+
+    SELECT json_array() --> '[]'
+
+Null handling
+^^^^^^^^^^^^^
+
+If a value passed for an array element is ``null``, it is treated according to
+the specified null treatment option. If ``ABSENT ON NULL`` is specified, the
+null element is omitted in the result. If ``NULL ON NULL`` is specified, JSON
+``null`` is added to the result. ``ABSENT ON NULL`` is default.
+
+.. code-block:: text
+
+    SELECT json_array(true, null, 1)                --> '[true,1]'
```

Comment:
should we put result in next line and add empty line after like in other snippets below .. it make it nicer to read and to cut and paste from..